### PR TITLE
Add alks_session data source

### PR DESCRIPTION
**Pull request by jcarlson**
## Change Log Items

* Adds a new data source provider, `alks_session`

## Description

The new data source, `alks_session`, surfaces the AWS STS credentials that the `alks` provider is configured with

### CODE REVIEW

**Comment by stephenleemcfarland** [comment]
This approach is much more in-line with how I would expect a datasource to behave. Maybe we should mark the other one as deprecated? Thanks for this contribution @jcarlson ! We should also add a doc [here](https://github.com/Cox-Automotive/terraform-provider-alks/tree/master/docs/data-sources) so that when published as an official tf provider people will get some documentation to go with it.

File: data_source_alks_session.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+import (
+	"context"
+	"github.com/Cox-Automotive/alks-go"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceAlksSession() *schema.Resource {
```
Comment (jcarlson) [inline data_source_alks_session.go:11]: @stephenleemcfarland Do you think `alks_session` is the right name for this data source? Would `alks_credentials` or `alks_session_credentials` be more clear?

**Comment by jcarlson** [comment]
@stephenleemcfarland Documentation added.

File: data_source_alks_session.go (hunk 1 above)
Comment (stephenleemcfarland) [reply data_source_alks_session.go:11]: maybe `aws_session_credentials` cause that's what it really is? I don't feel strongly about this. I think your current name is reasonable already

**Review by stephenleemcfarland — APPROVED** [review]
This looks great! I will try to test out locally myself but this looks as expected. Thanks again for the contribution @jcarlson

**Comment by DevOpsDave** [comment]
@jcarlson This is under review.  We will get back to you asap with a merge or comments.

**Comment by jcarlson** [comment]
@DevOpsDave Thanks. Here is [an example](https://ghe.coxautoinc.com/logistics-sre/spikes/pull/16/commits/31e9b6ba75100f8144872a8fb25efba4f024a70d?diff=unified&w=0#diff-d780168c11dbd50e695326e57df0b86a541fbed21831752d29e5990eeeb51217R25-R51) of how this new data source would be used, but substituting the `data "alks_keys" "session"` with `data "alks_session" "current"`.

It's worth noting that, like all `data` objects, all of the output attributes from this `data` object are stored in Terraform state.

File: data_source_alks_session.go [hunk 2]
```diff
@@ -33,13 +36,19 @@ func dataSourceAlksSessionRead(ctx context.Context, d *schema.ResourceData, meta
...
 	if ok {
 		_ = d.Set("access_key", stsCredentials.AccessKey)
 		_ = d.Set("secret_key", stsCredentials.SecretKey)
 		_ = d.Set("session_token", stsCredentials.SessionToken)
+		// When the below line was not present I could not get credentials to be returned
+		// as terraform outputs during runtime.
+		d.SetId(client.AccountDetails.Account)
 	}
 
+	// If not ok, we return empty values for credentials and move along.  Is this what we want?
```
Comment (jcarlson) [inline data_source_alks_session.go:51]: I would think best to fail fast, if possible. Whatever you’re using these credentials for downstream is not going to work if this outputs blank values, so raising an exception here might help triangulate the problem faster. I just did not know how to raise errors in Go/Terraform providers.

**Comment by DevOpsDave** [comment]
> @DevOpsDave Thanks. Here is [an example](https://ghe.coxautoinc.com/logistics-sre/spikes/pull/16/commits/31e9b6ba75100f8144872a8fb25efba4f024a70d?diff=unified&w=0#diff-d780168c11dbd50e695326e57df0b86a541fbed21831752d29e5990eeeb51217R25-R51) of how this new data source would be used, but substituting the `data "alks_keys" "session"` with `data "alks_session" "current"`.
> 
> It's worth noting that, like all `data` objects, all of the output attributes from this `data` object are stored in Terraform state.

I believe my update designates the secret key and session key as sensitive so their plain text values might not show up there.  Gonna verify.

**Comment by DevOpsDave** [comment]
> > @DevOpsDave Thanks. Here is [an example](https://ghe.coxautoinc.com/logistics-sre/spikes/pull/16/commits/31e9b6ba75100f8144872a8fb25efba4f024a70d?diff=unified&w=0#diff-d780168c11dbd50e695326e57df0b86a541fbed21831752d29e5990eeeb51217R25-R51) of how this new data source would be used, but substituting the `data "alks_keys" "session"` with `data "alks_session" "current"`.
> > It's worth noting that, like all `data` objects, all of the output attributes from this `data` object are stored in Terraform state.
> 
> I believe my update designates the secret key and session key as sensitive so their plain text values might not show up there. Gonna verify.

No that is not the case.  Just verified.

**Comment by DevOpsDave** [comment]
@jcarlson , closing this PR out.  Ping me on slack if I can help with creating a new tfp that will fit for you.

**Pull request closed without merging**